### PR TITLE
Versioning & Release ADR

### LINKED ISSUE #25: Write ADR on versioning & release workflow for portal codebase and supporting infrastructure

### PR DESCRIPTION
**Pull request by suzubara**
## Description 

This is an ADR that describes our development to deployment workflow, and versioning / release pipeline.

Fixes #25 

## Review Notes

You can view the draft ADR here: https://github.com/USSF-ORBIT/ussf-portal/blob/b5bb74e6ecb7aa55c9f3747cdd99b58157c7592b/docs/adr/0003-versioning-releasing.md

As you can see there is not a decided option yet! Please chime in with your thoughts :)

### CODE REVIEW

File: docs/adr/0003-versioning-releasing.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+## Decision Drivers
+
+- It's important to us to maintain some version number that is easy to communicate and represents the code at a given point in time.
+- We want to automate as much of the development -> production pipeline as possible, with the exception of a single manual gate we can use to determine whether a set of changes should go to production.
+
+## Considered Options
+
+- Use the model known as [Git-flow](https://nvie.com/posts/a-successful-git-branching-model/)
+- Use the model known as [GitHub flow](https://guides.github.com/introduction/flow/)
```
Comment (noahfirth) [inline docs/adr/0003-versioning-releasing.md:19]: My vote would be for GitHub flow (though as someone who won't likely have a lot of actual code writing time under my fingers, my vote should only be treated as a suggestion). Maintaining a single branch that is always available to be deployed in production moves away from a "versioned release" model that makes sense for deployed software but may not make as much sense in the context of a site.

While it does require being cognizant of and careful around feature addition, it feels like a simpler, more continuous workflow for what we are doing.

Comment (suzubara) [reply docs/adr/0003-versioning-releasing.md:19]: @noahfirth We discussed during eng sync and have landed on Github flow as well!

**Comment by suzubara** [comment]
I updated the ADR & tried to capture all of the points made during our discussion, but please comment if I missed anything!

**Review by abbyoung — APPROVED** [review]
Beautiful! Thanks for capturing so much detail and our eng sync discussion.

**Review by esacteksab — APPROVED** [review]

**Pull request merged**